### ISSUE TITLE
Can it be used for information retrieval o embedding generator for long text ? 

### ISSUE THREAD

**Issue by wilfoderek** [abertsch72/unlimiformer#67]

**Comment by abertsch72** [comment]
Hey, thanks for your interest and apologies for the late reply! As Unlimiformer incorporates the long context on the *decoding* side, it cannot naively be applied for embedding generation-- however, it's possible this or a similar method could be adapted to that use case by using hidden states from the decoder (after several generation steps) as the embeddings.

**Comment by wilfoderek** [comment]
Thank you for your response! Could you possibly provide an example of how to achieve this?
My idea is to train Unlimiformer on short queries and long texts to generate embeddings effectively. I would greatly appreciate any guidance or insights you can share on this approach.



Sincerely yours,
  Wilfo Ivan Martel S.
 Web Developer -  Cell Phone:0982502692
 Blog: http://neurocodigo.wordpress.com
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
"La religión ni las leyes del estado definen el comportamiento de un ser humano , eso depende de la moral y la conciencia de cada uno de nosotros."
Sócrates.



________________________________

**Comment by abertsch72** [comment]
Yeah, definitely! I think you would want to train on some task where you'd expect the outputs to use the full inputs-- if you have the right data, document summarization would be a great choice. Then to generate embeddings, you'd pass an input, generate $n$ of tokens of summary (which you can discard), and after $n$ tokens generated take the decoder hidden state as your embedding. (This is different from the usual setting, where you'd take the encoder hidden state to be the embedding.) Setting $n$ higher allows the model (on average) to see more of the inputs, since different parts of the input will be retrieved during each decoding step / at each head; but this adds up quickly. I haven't tried anything like this, but I imagine using even $n=30$ or so might be sufficient.
